todo.js

```diff
@@ -124,7 +114,53 @@ function updateTable() {
 
 // Initialize table with current month
 window.onload = function () {
-  //const currentMonth = new Date().getMonth();
-  //document.getElementById('monthSelector').value = currentMonth;
-  createTable();
+  createTable(state.todoList);
+};
+
+//Filter stored tomatoes by id, year and month
+const currentTomatoes = (id) => {
+  return state.timer.tomatoes.filter((todo) => isThisCurrentTodo(todo, id));
+};
+
+//Filter stored tomatoes by day
+const currentTomatoesByDay = (id, day) => {
+  return state.timer.tomatoes.filter((todo) => isThisCurrentDay(todo, id, day));
+};
+
+//Check if the current day is the same as the day of the todo
+function isThisCurrentDay(todo, id, day) {
+  return todo.id === id && todo.start.year === state.timer.currentYear && todo.start.month === state.timer.currentMonth && todo.start.day === day;
 };
+
+//Check if the current todo is the same as the id of the todo and year and month
+function isThisCurrentTodo(todo, id) {
+  return todo.id === id && todo.start.year === state.timer.currentYear && todo.start.month === state.timer.currentMonth;
+}
+//Update the time of the tomato by day in table
+function updateTomatoTimeByDay(id, day) {
```

Comment:
it looks like this function doesn't really "update" the page, it just returns some text? then I would name it "getUpdatedTomatoTimeByDay" 
```suggestion
function getUpdatedTomatoTimeByDay(id, day) {
```

Reply:
and from this
![image](https://github.com/antondurinin/ToDo/assets/167963627/7d347425-e532-42c2-8ede-1dd96f590727)
by AI I made this
![image](https://github.com/antondurinin/ToDo/assets/167963627/0fb1e22f-b342-4583-9133-733f8a950ac5)

Reply:
Cool, actually looking at this again, I notice we are storing all units of a date time (month day year etc) but is it necessary? They are all encoded inside the timestamp already right? 

What if we just store the timestamp instead? All those units can be derived from the timestamp, and then our state becomes much smaller

"Don't sync state, derive it!" Now this example is using React but the idea is the same
https://kentcdodds.com/blog/dont-sync-state-derive-it